substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/SharedLibraryViaCCBootImage.java

```diff
@@ -54,6 +55,9 @@ public SharedLibraryViaCCBootImage(HostedUniverse universe, HostedMetaAccess met
     @Override
     public LinkerInvocation write(DebugContext debug, Path outputDirectory, Path tempDirectory, String imageName, BeforeImageWriteAccessImpl config) {
         LinkerInvocation inv = super.write(debug, outputDirectory, tempDirectory, imageName, config);
+        if (NativeImageOptions.ExitAfterRelocatableImageWrite.getValue()) {
+            return inv;
+        }
         writeHeaderFiles(outputDirectory, imageName, inv.getSymbolAliases(), false);
```

Comment:
Have you considered passing `Collections.emptyMap()` in case when `inv == null`?